org.eclipse.jdt.launching/launching/org/eclipse/jdt/internal/launching/DetectVMInstallationsJob.java

```diff
@@ -0,0 +1,121 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.jdt.internal.launching;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.core.runtime.jobs.Job;
+import org.eclipse.jdt.launching.IVMInstall;
+import org.eclipse.jdt.launching.IVMInstallType;
+import org.eclipse.jdt.launching.JavaRuntime;
+import org.eclipse.jdt.launching.VMStandin;
+import org.eclipse.osgi.util.NLS;
+
+class DetectVMInstallationsJob extends Job {
+
+	private static final Object FAMILY = "Family " + DetectVMInstallationsJob.class.getName(); //$NON-NLS-1$
+	
+	DetectVMInstallationsJob() {
+		super(LaunchingMessages.lookupInstalledJVMs);
+	}
+
+	@Override
+	protected IStatus run(IProgressMonitor monitor) {
```

Comment:
The method is long.
Can we please split it into:
1) get known VM's
2) get candidate VM's
3) create new VM's for those from 2) who are not in 1)

It would be easier to read.

Reply:
Ack